moped-editor/src/utils/activityLogFormatters/mopedFundingActivity.js

```diff
@@ -0,0 +1,61 @@
+import MonetizationOnOutlinedIcon from "@material-ui/icons/MonetizationOnOutlined";
+import { ProjectActivityLogTableMaps } from "../../views/projects/projectView/ProjectActivityLogTableMaps";
+import { isEqual } from "lodash";
+
+export const formatFundingActivity = (
+  change,
+  fundingSources,
+  fundingPrograms
+) => {
+  const entryMap = ProjectActivityLogTableMaps["moped_proj_funding"];
+
+  const changeIcon = <MonetizationOnOutlinedIcon />;
+  let changeDescription = "Project funding updated";
+  let changeValue = "";
+
+  // add a new funding source
+  if (change.description.length === 0) {
+    changeDescription = "Added a new funding source ";
+    // if the added record has a funding source, use that as the change value
+    if (change.record_data.event.data.new.funding_source_id) {
+      changeValue =
+        fundingSources[change.record_data.event.data.new.funding_source_id];
+    // if not, then check if theres a funding program
+    } else if (change.record_data.event.data.new.funding_program_id) {
+      changeValue =
+        fundingPrograms[change.record_data.event.data.new.funding_program_id];
+    }
+    // if there isnt a funding source or program added, then default to ""
+    return { changeIcon, changeDescription, changeValue };
+  }
```

Comment:
When I was testing this, I was confused for a little while trying to figure out where the bold word being shown in the activity log for a new funding source entry was coming from. Being able to do this custom logic per activity log field and per action demonstrates the flexibility you've built into this. It's really nice.